### ISSUE TITLE
Wrap all errors related to file input to reference line numbers and row data

### ISSUE THREAD

**Issue by hepcat72** [Princeton-LSI-ResearchComputing/tracebase#646]
### Inspiration

Work on #580 to create errors that gave users useful information uncovered a recurring theme that could be applied to any exception.

### Description

When any exception is caught, and a row index and data is available, the exception should be wrapped in another exception that adds row number and data.  This was done for a number of specific exceptions, which were eventually moved outside of the catch and into the code (so that errant rows could be skipped for further processing.  Examples include `MissingCompound`, `MissingSamples`, `ConflictingValueError`, etc.  Some of these exception classes take data on multiple rows.  Others take a single row.  An exception class could be created for either case or one class could handle either case.

The class would need to take (for example) a dict with row data, (a) row number(s), and either multiple or a single caught exception's message.

### Alternatives

none

### Dependencies

none

### Comment

The utility of this exception class would be to add context to unanticipated exceptions and could be converted into custom exceptions that add suggestions on how to resolve the error.  Each time we investigate a certain type of exception, we are learning how to fix errors of that type.  That learned knowledge should be recorded in the form of a custom exception class to give the next curator a jump start on dealing with that issue the next time it arises.

-----

### Assumptions

- List of assumptions that the code will not explicitly address/check
- E.g. We will assume input is correct (explaining why there is no validation)

### Requirements

- [ ] 1. List of numbered conditions to be met for the feature
- [ ] 2. E.g. Every column/row must display a value, i.e. cannot be empty
- [ ] 3. Numbers for reference & checkboxes for progress tracking

### Limitations

- A list of things this work will specifically not do
- E.g. This feature will only handle the most frequent use case X

### Affected Components

A tentative list of anticipated repository items that will be changed, labeled
with "add", "delete", or "change".  One item per line.  (Mostly, this will be a
list of files.)

- change: File path or DB table ...
- add: Environment variable or server setting
- delete: External executable or cron job

#### Interface Change description

Describe changes to usage. *E.g. GUI/command-line changes*

#### Code Change Description

Describe code changes planned for the feature. *(Pseudocode encouraged)*

#### Tests

- [ ] 1. A description of at least one test for each requirement above.
- [ ] 2. E.g. Test for req 2 that there's an exception when display value is ''
- [ ] 3. Numbers for reference & checkboxes for progress tracking

**Comment by hepcat72** [comment]
Created a `FileLoadError` exception class.

**Comment by hepcat72** [comment]
This is essentially done (though not utilized everywhere yet).  I handled this by adding a method to `InfileError` called `set_formatted_messages`, which changes self.message.  I also added `__str__()` which returns `self.message`.  The exception thus isn't repackaged.  Its content is just changed to include the file location details.  And as for multiple row numbers in those group exceptions, I created exception classes for summarizing exceptions that take lists of exceptions and organize the data using dicts.
